### PR TITLE
feat(chore): unsplit fiches-mt

### LINKED ISSUE #1394: Arrêter le split des fiches MT
To do :
- regrouper les fiches actuelles
- afficher les anciennes fiches sous forme d'accordéons dans la nouvelle fiche regroupée
- les thémer grâce au data-filler
- impact sur la recherche : voir comment conserver une indexation correcte de tout le contenu

### PR DESCRIPTION
**Pull request by UnbearableBear**
closes #1394

A few things to note:

- I created a new index to store the complete fiches mt
- They are still fiche-mt documents in the main index but they are split (as before) and have the same slug by parent fiche (refering to the complete fiches-mt in the index mentionned above), except they also have a #id appended at their end.
- Now documents' slug have an `asKeyword` field which is the same slug but with a keyword type (so that you can filter on them instead of matching).
- The script adding themes to fiches MT will need an update in the datafiller so that new slugs are taken into account.
- Datafiller will also need an update to handle additional hashes appended at the end of some slugs
- They are no more linked contents in fiche-sp for the moment due to this refactor but we shall add them back later on (@lionelB  is refactoring this part so I prefer not to touch it for now)
- They now do are accordions in the fiches MT and they are opened automatically if the correct hash is provided in the url (no on SSR though)

Other lines of thought:

- https://blog.mimacom.com/parent-child-elasticsearch/ -> Interesting read but I'm not sure we need to optimise to this extent and this implies to keep everything on the same indice which also means to modify the nlp code -> we loose a little separation of concerns
- Consider also splitting fiches-sp in the same way to improve search results

### CODE REVIEW

**Comment by lionelB** [comment]
> Previously, slugs where not considered as keywords in the document indice which is now the case. This makes it more consistent across indices and make the search by slug very accurate

slug being keyword was the case originally, but where chenged to text in order to make match request for datafiller content, since some url are changing way too often. Usign allow to search and fix cc @revolunet

**Comment by lionelB** [comment]
note: merging this PR should be sync with an update of the datafiller

**Comment by UnbearableBear** [comment]
> > Previously, slugs where not considered as keywords in the document indice which is now the case. This makes it more consistent across indices and make the search by slug very accurate
> 
> slug being keyword was the case originally, but where chenged to text in order to make match request for datafiller content, since some url are changing way too often. Usign allow to search and fix cc @revolunet

Oh I didn't know that ! This strategy makes it possible for wrong documents to be referenced in the long term, I'm not fond of it but I can also adapt my code to make everything work with text slugs instead of keyword

File: packages/code-du-travail-api/src/server/routes/sheets-mt/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Return the sheet-mt that match a given slug
+ *
+ * @example
+ * http://localhost:1337/api/v1/sheets-mt/:slug
+ *
+ * @returns {Object} An object containing the matching sheet-mt .
+ */
+
+router.get("/sheets-mt/:slug", async ctx => {
```
Comment (revolunet) [inline packages/code-du-travail-api/src/server/routes/sheets-mt/index.js:21]: ce serait la seule route en anglais... on pourrait rester en fr par consistance ?

Comment (UnbearableBear) [reply packages/code-du-travail-api/src/server/routes/sheets-mt/index.js:21]: ah je l'ai mise en anglais justement parce qu'il me semblait que toutes les autres routes étaient en anglais ! (search / items / docs-count etc.)

Comment (revolunet) [reply packages/code-du-travail-api/src/server/routes/sheets-mt/index.js:21]: ure right :)

**Comment by revolunet** [comment]
joli change qui va p-e être utile à @nkrmr pour l'intégration des fiches dans les contributions

File: packages/code-du-travail-api/tests/fiches_ministere_travail_data.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+[{
+  "date": "23/05/2017",
```
Comment (lionelB) [inline packages/code-du-travail-api/tests/fiches_ministere_travail_data.json]: once rebased, this document should be generated using the script that update mock data

File: packages/code-du-travail-data/dataset/fiches_ministere_travail/index.js [hunk 3]
```diff
@@ -147,11 +120,29 @@ async function parseFiche(url) {
...
+      })),
+      null,
+      2
+    )
+  );
   const fiches = results
     .map(splitArticle)
-    .reduce((state, documents) => state.concat(documents), []);
+    .reduce((accumulator, documents) => accumulator.concat(documents), []);
+  fs.writeFileSync(
```
Comment (lionelB) [inline packages/code-du-travail-data/dataset/fiches_ministere_travail/index.js:141]: console.log / console.error please :)

File: packages/code-du-travail-data/dataset/fiches_ministere_travail/package.json [hunk 4]
```diff
@@ -8,15 +8,17 @@
...
     "jsdom": "^15.2.1",
     "ora": "^4.0.2",
     "p-limit": "^2.2.1",
     "superagent": "^5.1.0"
   },
   "scripts": {
-    "prestart": "node verifyUrl.js",
-    "start": "node index.js > fiches-mt.json",
-    "updateTheme": "node updateTheme.js > out.json && mv out.json fiches-mt.json && node report.js"
+    "verifyUrl": "node verifyUrl",
```
Comment (lionelB) [inline packages/code-du-travail-data/dataset/fiches_ministere_travail/package.json]: why dit you remove it from prestart ? it on pupise that this script is called before updating content so we can know about potential url changes

File: packages/code-du-travail-data/dataset/fiches_ministere_travail/report.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
-const fiches = require("./fiches-mt.json");
+const fs = require("fs");
+const fiches = JSON.parse(fs.readFileSync("./fiches-mt.json"));
```
Comment (lionelB) [inline packages/code-du-travail-data/dataset/fiches_ministere_travail/report.js]: why this change ?

File: packages/code-du-travail-data/dataset/fiches_ministere_travail/updateTheme.js [hunk 6]
```diff
@@ -25,14 +27,9 @@ async function updateTheme(fiches) {
...
-  updateTheme(fiches);
-  console.log(JSON.stringify(fiches, null, 2));
-}
-
-if (module === require.main) {
-  main();
-}
-
-module.exports = updateTheme;
+let fiches = JSON.parse(fs.readFileSync("./fiches-mt.json"));
```
Comment (lionelB) [inline packages/code-du-travail-data/dataset/fiches_ministere_travail/updateTheme.js]: why didn't  you  require json file ?

File: packages/code-du-travail-data/indexing/populate.js [hunk 7]
```diff
@@ -114,33 +114,11 @@ function* cdtnDocumentsGen() {
...
-      intro,
-      description,
-      text,
-      html,
-      breadcrumbs,
-      theme,
-      date,
-      url,
-      anchor
+      ...fiche
```
Comment (lionelB) [inline packages/code-du-travail-data/indexing/populate.js]: I would prefer not use rest params here since it hides all the props

File: packages/code-du-travail-frontend/pages/fiche-ministere-travail/[slug].js [hunk 8]
```diff
@@ -78,3 +89,12 @@ const Intro = styled(Html)`
     margin-bottom: 0;
   }
 `;
+
+const TabContent = styled(Html)`
```
Comment (lionelB) [inline packages/code-du-travail-frontend/pages/fiche-ministere-travail/[slug].js:93]: should this be in @socialgouv/react-fiche-service-public ?

File: packages/code-du-travail-frontend/src/search/SearchResults/Results.js [hunk 9]
```diff
@@ -37,13 +37,21 @@ const ListLink = ({
     );
   }
 
+  let rootSlug = slug;
+  let anchor = "";
+  if (slug.includes("#")) {
+    [rootSlug, anchor] = slug.split("#");
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/search/SearchResults/Results.js:43]: using a default value should avoid ternary in `as`

File: packages/react-ui/src/Accordion/index.js [hunk 10]
```diff
@@ -14,30 +14,36 @@ import { fadeIn } from "../keyframes";
 
 export const Accordion = ({ items, ...props }) => (
   <RootAccordion allowZeroExpanded allowMultipleExpanded {...props}>
-    {items.map((item, index) => (
-      <StyledAccordionItem key={index}>
+    {items.map(({ body, id, title }, index) => (
+      <StyledAccordionItem uuid={id} key={index}>
```
Comment (lionelB) [inline packages/react-ui/src/Accordion/index.js:18]: uuid ?

File: packages/react-ui/src/Accordion/index.js [hunk 11]
```diff
@@ -14,30 +14,36 @@ import { fadeIn } from "../keyframes";
...
-        <StyledAccordionItemPanel>{item.body}</StyledAccordionItemPanel>
+        <StyledAccordionItemPanel>{body}</StyledAccordionItemPanel>
       </StyledAccordionItem>
     ))}
   </RootAccordion>
 );
 
 Accordion.propTypes = {
   className: PropTypes.string,
+  preExpanded: PropTypes.arrayOf(PropTypes.string),
```
Comment (lionelB) [inline packages/react-ui/src/Accordion/index.js:33]: like the idea to use an array :)

File: packages/code-du-travail-data/dataset/fiches_ministere_travail/index.js (hunk 3 above)
Comment (UnbearableBear) [reply packages/code-du-travail-data/dataset/fiches_ministere_travail/index.js:141]: J'écris aussi dans deux fichiers ici !

File: packages/code-du-travail-frontend/pages/fiche-ministere-travail/[slug].js (hunk 8 above)
Comment (UnbearableBear) [reply packages/code-du-travail-frontend/pages/fiche-ministere-travail/[slug].js:93]: En fait le component `Html` rajoute une div et ça annule le selecteur d'ou le rajout ici, parce qu'il y est déjà dans UI

File: packages/code-du-travail-api/src/server/routes/items/getRelatedItems.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+      { index },
+      // we +1 the size to remove the document source that should match perfectly for the given vector
+      { ...getSemBody({ query_vector: queryVector, size: size + 1, sources }) }
+    ]
+  });
+
+  const { hits: { hits: semanticHits } = { hits: [] } } = semResponse;
+  const { hits: { hits: fullTextHits } = { hits: [] } } = esResponse;
+
+  console.log(JSON.stringify(fullTextHits, null, 2));
```
Comment (nkrmr) [inline packages/code-du-travail-api/src/server/routes/items/getRelatedItems.js]: Remove console.log

**Review by nkrmr — APPROVED** [review]

File: packages/code-du-travail-frontend/src/common/__tests__/Suggester.test.js [hunk 13]
```diff
@@ -26,6 +26,9 @@ describe("<Suggester />", () => {
     expect(container).toMatchSnapshot();
   });
 
+  /* eslint-disable */
+
+  /*
```
Comment (lionelB) [inline packages/code-du-travail-frontend/src/common/__tests__/Suggester.test.js]: o_O

Comment (UnbearableBear) [reply packages/code-du-travail-frontend/src/common/__tests__/Suggester.test.js]: Alors petites précisions tout de même: 
1) Ce sont des test sans assertion, le linter rale à cause de ça. (c'est vrai que des tests qui vérifient rien c'est pas très utile ceci dit
2) Le linter râle si je commente de tests
3) Je n'ai aucune idée de ce qui a voulu être testé ici et je ne pense pas que ce soit dans cette PR qu'il faut le faire

Comment (lionelB) [reply packages/code-du-travail-frontend/src/common/__tests__/Suggester.test.js]: c'est des warning pas des erreurs. Pour rappel les getText, getByLabelText, ... de la lib permette de faire des assertions sur le contenu donc, oui il y a des assertions mais pas au sens expect de jest.

Comment (lionelB) [reply packages/code-du-travail-frontend/src/common/__tests__/Suggester.test.js]: il faudra décommenté

**Comment by revolunet** [comment]
Je remarque quelques soucis dans les anchors : 

 - nbsp tronqué : "Quelles-sanctions-sont-applicables-si-le-SMIC-n-est-pas-atteint-nbsp"
 - pb avec apostrophe : "A-quel-conseil-de-prud-039-hommes-s-039-adresser"

Est-ce que c'est fixable de notre côté ?

**Comment by UnbearableBear** [comment]
https://travail-emploi.gouv.fr/droit-du-travail/la-remuneration/article/le-smic#Quelles-sanctions-sont-applicables-si-le-SMIC-n-est-pas-atteint-nbsp

https://travail-emploi.gouv.fr/droit-du-travail/le-reglement-des-conflits-individuels-et-collectifs/article/le-conseil-de-prud-hommes#A-quel-conseil-de-prud-039-hommes-s-039-adresser

On récupère leur id vu qu'on affiche directement leur html (pour que le anchor marche). On pourrait fixer ça de notre côté mais ça ferait du code pas beau et bien trop spécifique. A mon avis il vaut mieux le laisser tel quel pour le moment, ça reste un cas assez rare

**Comment by revolunet** [comment]
ok !
